pynsee/utils/init_conn.py

```diff
@@ -84,30 +90,28 @@ def init_conn(
             f"locally here:\n{config_file}"
         )
 
-    # save config
-    if not invalid_requests and ("Sirene" not in invalid_requests):
-        # everything works, or at least SIRENE
-        config = {
-            "sirene_key": sirene_key,
-            "http_proxy": http_proxy,
-            "https_proxy": https_proxy,
-        }
+    if invalid_requests and ("Sirene" in invalid_requests):
 
+        if not sirene_key:
+            # user has probably no use for SIRENE API and has already
+            # been warned about this
+            sirene_key = None
+        else:
+            # invalid sirene key, restore from previous valid stored key
+            try:
+                sirene_key = init_config["sirene_key"]
+            except KeyError:
+                sirene_key = None
+            else:
+                logger.warning("Previous SIRENE key has been restored.")
+
+    config = {
+        "sirene_key": sirene_key,
+        "http_proxy": http_proxy,
+        "https_proxy": https_proxy,
+    }
+
+    if not invalid_requests:
```

Comment:
I thought you wanted to save if sirene was the only issue in the api?

If so, use ``if set(invalid_requests).issubset({"Sirene"})`` or ``if {"Sirene"}.issuperset(invalid_requests)``.